# DSCI 6002 Midterm Project Code (Team 8)
### Moureen Nampijja, Smit Patel, Mahesh Potu, Jafar Vohra

We will start by importing the necessary libraries and modules to help us complete this portion of the project...

In [1]:
import pandas as pd
import matplotlib.pyplot as plt

Since our data is uploaded to our team GitHub page, we will use pandas to read the CSV data via the URL of the raw data. The first five rows of the data are then able to be seen below...

In [2]:
data = pd.read_csv("heart_attack_prediction_dataset.csv")
data.head()

Patient ID  Age     Sex  Cholesterol Blood Pressure  Heart Rate  Diabetes  \
0    BMW7812   67    Male          208         158/88          72         0   
1    CZE1114   21    Male          389         165/93          98         1   
2    BNI9906   21  Female          324         174/99          72         1   
3    JLN3497   84    Male          383        163/100          73         1   
4    GFO8847   66    Male          318          91/88          93         1   

   Family History  Smoking  Obesity  ...  Sedentary Hours Per Day  Income  \
0               0        1        0  ...                 6.615001  261404   
1               1        1        1  ...                 4.963459  285768   
2               0        0        0  ...                 9.463426  235282   
3               1        1        0  ...                 7.648981  125640   
4               1        1        1  ...                 1.514821  160555   

         BMI  Triglycerides  Physical Activity Days Per Week  \
0  31.251233            286                                0   
1  27.194973            235                                1   
2  28.176571            587                                4   
3  36.464704            378                                3   
4  21.809144            231                                1   

   Sleep Hours Per Day    Country      Continent           Hemisphere  \
0                    6  Argentina  South America  Southern Hemisphere   
1                    7     Canada  North America  Northern Hemisphere   
2                    4     France         Europe  Northern Hemisphere   
3                    4     Canada  North America  Northern Hemisphere   
4                    5   Thailand           Asia  Northern Hemisphere   

   Heart Attack Risk  
0                  0  
1                  0  
2                  0  
3                  0  
4                  0  

[5 rows x 26 columns]

Let's check out the shape of the data that we have just loaded in...

In [3]:
data.shape

(8763, 26)

From the above output, we can see that there are 8,763 rows of patient data with 26 columns (or features) of data related to each patient. Let's get an idea of what features are provided with this dataset...

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8763 entries, 0 to 8762
Data columns (total 26 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   Patient ID                       8763 non-null   object 
 1   Age                              8763 non-null   int64  
 2   Sex                              8763 non-null   object 
 3   Cholesterol                      8763 non-null   int64  
 4   Blood Pressure                   8763 non-null   object 
 5   Heart Rate                       8763 non-null   int64  
 6   Diabetes                         8763 non-null   int64  
 7   Family History                   8763 non-null   int64  
 8   Smoking                          8763 non-null   int64  
 9   Obesity                          8763 non-null   int64  
 10  Alcohol Consumption              8763 non-null   int64  
 11  Exercise Hours Per Week          8763 non-null   float64
 12  Diet                

We can see a lot of information from the output of the above code. Let's break it down...

- Each feature shows a non-null count equivalent to the number of total entries, thus saying there are no null values in this dataset. Let's check this another way to be sure...

In [5]:
data.isna().sum()

Patient ID                         0
Age                                0
Sex                                0
Cholesterol                        0
Blood Pressure                     0
Heart Rate                         0
Diabetes                           0
Family History                     0
Smoking                            0
Obesity                            0
Alcohol Consumption                0
Exercise Hours Per Week            0
Diet                               0
Previous Heart Problems            0
Medication Use                     0
Stress Level                       0
Sedentary Hours Per Day            0
Income                             0
BMI                                0
Triglycerides                      0
Physical Activity Days Per Week    0
Sleep Hours Per Day                0
Country                            0
Continent                          0
Hemisphere                         0
Heart Attack Risk                  0
dtype: int64

- Let's also check to see if there are any duplicated records in this dataset...

In [6]:
data.duplicated().sum()

0

NO missing values and NO duplicated records in this dataset. This completes our initial data cleaning steps. Let's dive into each column and we will compile a list of features to clean / engineer for the further steps of this project...

- The first column is a unique identifier for each patient, called **Patient ID**. This data is read in from the CSV file as a string variable.

In [7]:
data["Patient ID"].nunique()

8763

- The last column is an indicator of **Heart Attack Risk** for each patient. This data is read in from the CSV file as an integer variable with values of 0 or 1.

In [8]:
data["Heart Attack Risk"].value_counts()

0    5624
1    3139
Name: Heart Attack Risk, dtype: int64

In [9]:
### VISUALIZE THE HEART ATTACK RISK DATA HERE

- The second column provides data for the **Age** of the patient. This is read in from the CSV file as an integer variable.

In [10]:
### VISUALIZE THE AGE DATA HERE

- The third column is an identifier for the **Sex** of the patient, either Male or Female. This is read in from the CSV file as a string variable.

In [11]:
data["Sex"].value_counts()

Male      6111
Female    2652
Name: Sex, dtype: int64

In [12]:
### VISUALIZE THE SEX DATA HERE

- The fourth column provides data for the **Cholesterol**, measured in milligrams per deciliter (mg/dL). This is read in from the CSV file as an integer variable.

In [13]:
### VISUALIZE THE CHOLESTEROL DATA HERE

- The fifth column is a feature that provides the **Blood Pressure** of the patient, including the systolic and diastolic numbers. This is read in from the CSV file as a string variable, with the two number seperated by a "/".
- We will look to feature engineer this column in order to seperate the **Systolic Blood Pressure** and **Diastolic Blood Pressure** for use later in the project

In [14]:
data["Blood Pressure"].sample()

7373    141/109
Name: Blood Pressure, dtype: object

In [15]:
data[["Systolic Blood Pressure","Diastolic Blood Pressure"]] = data["Blood Pressure"].str.split("/", expand=True)
data[["Blood Pressure","Systolic Blood Pressure","Diastolic Blood Pressure"]].head()

Blood Pressure Systolic Blood Pressure Diastolic Blood Pressure
0         158/88                     158                       88
1         165/93                     165                       93
2         174/99                     174                       99
3        163/100                     163                      100
4          91/88                      91                       88

In [16]:
### VISUALIZE SYSTOLIC AND DIASTOLIC BLOOD PRESSURE

- The sixth column...

In [17]:
max_value = data['Heart Rate'].max()
min_value = data['Heart Rate'].min()
average_value = data['Heart Rate'].mean()

print("Average value: ", average_value)
print("Max value: ", max_value)
print("Min value: ", min_value)

Average value:  75.02168207234965
Max value:  110
Min value:  40


In [18]:
df_sorted=data.sort_values(by='Heart Rate')
print(df_sorted[['Heart Rate', 'Heart Attack Risk']])

      Heart Rate  Heart Attack Risk
802           40                  0
4007          40                  0
1277          40                  1
3639          40                  1
7110          40                  0
...          ...                ...
107          110                  0
4856         110                  0
974          110                  1
8690         110                  0
2185         110                  0

[8763 rows x 2 columns]


- The seventh column...

In [19]:
data["Diabetes"].value_counts()

1    5716
0    3047
Name: Diabetes, dtype: int64

- The eight column...

In [20]:
data["Family History"].value_counts()

0    4443
1    4320
Name: Family History, dtype: int64

- The ninth column...

In [21]:
data["Smoking"].value_counts()

1    7859
0     904
Name: Smoking, dtype: int64

- The tenth column...

In [22]:
data["Obesity"].value_counts()

1    4394
0    4369
Name: Obesity, dtype: int64

- The eleventh column...

In [23]:
data["Alcohol Consumption"].value_counts()

1    5241
0    3522
Name: Alcohol Consumption, dtype: int64

- The twelfth column...

In [24]:
max_value = data['Exercise Hours Per Week'].max()
min_value = data['Exercise Hours Per Week'].min()
average_value = data['Exercise Hours Per Week'].mean()

print("Average value: ", average_value)
print("Max value: ", max_value)
print("Min value: ", min_value)

Average value:  10.01428365521483
Max value:  19.998709051535457
Min value:  0.0024423483189783


- The thirteenth column provides the type of **Diet** the patient has, including Healthy, Unhealthy, and Average classifications. This is read in from the CSV file as a string variable.

In [25]:
data["Diet"].value_counts()

Healthy      2960
Average      2912
Unhealthy    2891
Name: Diet, dtype: int64

In [26]:
### VISUALIZE THE DIET DATA HERE

- The fourteenth column...

In [27]:
data["Previous Heart Problems"].value_counts()

0    4418
1    4345
Name: Previous Heart Problems, dtype: int64

- The fifteenth column...

In [28]:
data["Medication Use"].value_counts()


0    4396
1    4367
Name: Medication Use, dtype: int64

- The sixteenth column...

In [29]:
max_value = data['Stress Level'].max()
min_value = data['Stress Level'].min()
average_value = data['Stress Level'].mean()

print("Average value: ", average_value)
print("Max value: ", max_value)
print("Min value: ", min_value)

Average value:  5.469702156795618
Max value:  10
Min value:  1


- The seventeenth column...

In [30]:
max_value = data['Sedentary Hours Per Day'].max()
min_value = data['Sedentary Hours Per Day'].min()
average_value = data['Sedentary Hours Per Day'].mean()

print("Average value: ", average_value)
print("Max value: ", max_value)
print("Min value: ", min_value)


Average value:  5.993690145051085
Max value:  11.999313410370352
Min value:  0.0012632057782457


In [31]:
df_sorted=data.sort_values(by='Sedentary Hours Per Day')
print(df_sorted[['Sedentary Hours Per Day', 'Heart Attack Risk']])

      Sedentary Hours Per Day  Heart Attack Risk
3314                 0.001263                  0
865                  0.001529                  1
2356                 0.003625                  0
970                  0.008307                  0
1915                 0.010631                  0
...                       ...                ...
4885                11.985484                  1
5032                11.987716                  0
7101                11.989217                  1
844                 11.992341                  0
1308                11.999313                  1

[8763 rows x 2 columns]


- The eighteenth column...

In [32]:
max_value = data['Income'].max()
min_value = data['Income'].min()
average_value = data['Income'].mean()

print("Average value: ", average_value)
print("Max value: ", max_value)
print("Min value: ", min_value)

Average value:  158263.18190117538
Max value:  299954
Min value:  20062


- The nineteenth column...

In [33]:
max_value = data['BMI'].max()
min_value = data['BMI'].min()
average_value = data['BMI'].mean()

print("Average value: ", average_value)
print("Max value: ", max_value)
print("Min value: ", min_value)

Average value:  28.891445877259635
Max value:  39.99721081557256
Min value:  18.002336577801902


- The twentieth column...

In [34]:
max_value = data['Triglycerides'].max()
min_value = data['Triglycerides'].min()
average_value = data['Triglycerides'].mean()

print("Average value: ", average_value)
print("Max value: ", max_value)
print("Min value: ", min_value)

Average value:  417.67705123816046
Max value:  800
Min value:  30


- The twenty-first column...

In [35]:
max_value = data['Physical Activity Days Per Week'].max()
min_value = data['Physical Activity Days Per Week'].min()
average_value = data['Physical Activity Days Per Week'].mean()

print("Average value: ", average_value)
print("Max value: ", max_value)
print("Min value: ", min_value)


Average value:  3.48967248659135
Max value:  7
Min value:  0


- The twenty-second column...

In [36]:
max_value = data['Sleep Hours Per Day'].max()
min_value = data['Sleep Hours Per Day'].min()
average_value = data['Sleep Hours Per Day'].mean()

print("Average value: ", average_value)
print("Max value: ", max_value)
print("Min value: ", min_value)

Average value:  7.0235079310738335
Max value:  10
Min value:  4


- The **Country**, **Continent**, and **Hemisphere** columns provide geographical data of the patients in the form of strings.

In [37]:
data["Country"].value_counts()

Germany           477
Argentina         471
Brazil            462
United Kingdom    457
Australia         449
Nigeria           448
France            446
Canada            440
China             436
New Zealand       435
Japan             433
Italy             431
Spain             430
Colombia          429
Thailand          428
South Africa      425
Vietnam           425
United States     420
India             412
South Korea       409
Name: Country, dtype: int64

In [38]:
### VISUALIZE THE COUNTRY DATA HERE

In [39]:
data["Continent"].value_counts()

Asia             2543
Europe           2241
South America    1362
Australia         884
Africa            873
North America     860
Name: Continent, dtype: int64

In [40]:
### VISUALIZE THE CONTINENT DATA HERE

In [41]:
data["Hemisphere"].value_counts()

Northern Hemisphere    5660
Southern Hemisphere    3103
Name: Hemisphere, dtype: int64

In [42]:
### VISUALIZE THE HEMISPHERE DATA HERE